app/parse.py

```diff
@@ -9,4 +14,26 @@
 
 
 def get_all_courses() -> list[Course]:
-    pass
+    results = []
+    request_courses = requests.get(BASE_URL)
+    soup = BeautifulSoup(request_courses.text, "html.parser")
+    courses = soup.select(".ProfessionCard_cardWrapper__JQBNJ")
+    for course in courses:
+        results.append(
+            Course(
+                name=course.select_one("a.typography_landingH3__vTjok").text,
+                short_description=course.select(
+                    "p.typography_landingTextMain__Rc8BD"
+                )[-1].text,
+                duration=course.select(
+                    ".ProfessionCard_professionTags__kCD1H >"
+                    " p.typography_landingTextMain__Rc8BD"
+                )[-1].text,
```

Comment:
Check the CSS selector '.ProfessionCard_professionTags__kCD1H > p.typography_landingTextMain__Rc8BD' for accuracy. Using [-1] assumes there is at least one matching element, which could lead to an IndexError if none are found.